lib/elixir/lib/regex.ex

```diff
@@ -186,20 +186,28 @@ defmodule Regex do
 
   defstruct re_pattern: nil, source: "", opts: [], re_version: ""
 
-  @type re_option ::
+  @type nl_spec() :: :cr | :crlf | :lf | :anycrlf | :any
+
+  @type compile_option ::
```

Comment:
Also renamed since it's been introduced in 1.17, and I think it's clearer + matches `:re`'s.

Reply:
If it was introduced in 1.17.0, maybe we should remove them for now. Erlang said they will change the RE engine for OTP 28, so any of those options may no longer be supported in a year. So perhaps we should remove those for now and backport?